`wit-bindgen-rust` failing to compile

Beginning this morning `wit-bindgen-rust` is failing to compile for me. This happens when my only dependency is `wit-bindgen` and it happens with both version `0.11.0` and `0.12.0` of that crate. Seems like it is likely related to the new version `0.12.0` of `wit-parser` published ~1 hour ago https://crates.io/crates/wit-parser

I am seeing both `0.11.3` and `0.12.0` pulled in
```sh
Compiling wit-parser v0.12.0
Compiling wit-parser v0.11.3
```
My Cargo.toml
```toml
[package]
name = "project"
version = "0.1.0"
edition = "2021"

[lib]
crate-type = ["cdylib"]

[dependencies]
wit-bindgen = { version = "0.11.0" }
```

Full error:
```sh
   Compiling wit-bindgen-rust v0.12.0
error[E0308]: arguments to this function are incorrect
   --> /Users/lnj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wit-bindgen-rust-0.12.0/src/lib.rs:394:30
    |
394 |         let component_type = wit_component::metadata::encode(
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
395 |             resolve,
    |             ------- expected `wit_parser::resolve::Resolve`, found `wit_bindgen_core::wit_parser::Resolve`
    |
    = note: `wit_bindgen_core::wit_parser::Resolve` and `wit_parser::resolve::Resolve` have similar names, but are actually distinct types
note: `wit_bindgen_core::wit_parser::Resolve` is defined in crate `wit_parser`
   --> /Users/lnj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wit-parser-0.11.3/src/resolve.rs:31:1
    |
31  | pub struct Resolve {
    | ^^^^^^^^^^^^^^^^^^
note: `wit_parser::resolve::Resolve` is defined in crate `wit_parser`
   --> /Users/lnj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wit-parser-0.12.0/src/resolve.rs:31:1
    |
31  | pub struct Resolve {
    | ^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `wit_parser` are being used?
note: expected `wit_parser::World`, found `wit_bindgen_core::wit_parser::World`
   --> /Users/lnj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wit-bindgen-rust-0.12.0/src/lib.rs:396:13
    |
396 |             world,
    |             ^^^^^
    = note: `wit_bindgen_core::wit_parser::World` and `wit_parser::World` have similar names, but are actually distinct types
note: `wit_bindgen_core::wit_parser::World` is defined in crate `wit_parser`
   --> /Users/lnj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wit-parser-0.11.3/src/lib.rs:251:1
    |
251 | pub struct World {
    | ^^^^^^^^^^^^^^^^
note: `wit_parser::World` is defined in crate `wit_parser`
   --> /Users/lnj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wit-parser-0.12.0/src/lib.rs:251:1
    |
251 | pub struct World {
    | ^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `wit_parser` are being used?
note: function defined here
   --> /Users/lnj/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wit-component-0.14.6/src/metadata.rs:152:8
    |
152 | pub fn encode(
    |        ^^^^^^

```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,8 +31,8 @@ indexmap = "2.0.0"
 
 wasm-encoder = "0.33.2"
 wasm-metadata = "0.10.6"
-wit-parser = "0.11.3"
-wit-component = "0.14.4"
+wit-parser = "0.12"
+wit-component = "0.14.6"
 
 wit-bindgen-core = { path = 'crates/core', version = '0.12.0' }
 wit-bindgen-c = { path = 'crates/c', version = '0.12.0' }

